src/waypoint.jsx

```diff
@@ -333,13 +352,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.noWrapper) {
+      const child = React.Children.only(this.props.children);
+      const ref = (node) => {
+        this.refElement(node);
+        if (this.props.children.ref) {
+          this.props.children.ref(node);
+        }
+      };
+      return React.cloneElement(child, { ref });
+    }
+
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
+  noWrapper: PropTypes.bool,
```

Comment:
I understand the pros of cons of each argument here. I like @trotzig 's conditional-wrapping solution, in that it tries to hide the complexity to the user. That can be useful for someone just getting started, as they have less to think about. @lencioni is probably right that at other times it could cause confusion when that same person doesn't understand why the div is added sometimes, but not others. I think the user still has full control here, because the situations where Waypoint wraps would always happen when it was absolutely necessary. It simply wouldn't work otherwise.

@lencioni 's suggestion to never wrap is appealing because I like giving all of the control to the developer, even if the learning curve is slightly higher at first. I think a helpful error message could quickly help someone get up to speed with when it's necessary, and when it's not. Also, avoiding another prop is a good thing, too. But I also understand where @rattrayalex is coming from about requiring folks to put wrapping divs everywhere. That is a definite downside to this approach.

I think I'd be alright with either magically wrapping or not wrapping ever, but I'm not a fan of a `noWrapper` prop.

If I had to pick between the options, it'd be @lencioni 's suggestion to not add the wrapping prop, and require the user wrap it as needed. As for the WaypointDiv component, I'd probably prefer to not include it just to keep this library simpler, but I don't feel too strongly about it.

tl;dr: as long as we avoid `noWrapper`, I'm OK with anything else.